geopandas/tests/test_geodataframe.py

```diff
@@ -843,6 +843,23 @@ def test_geodataframe_iterfeatures(self):
         result = list(df_only_numerical_cols.iterfeatures(na="keep"))[0]
         assert type(result["properties"]["Shape_Leng"]) is float
 
+        # When some features are non-scalar values
```

Comment:
Would you mind moving these tests into a new method, e.g. `test_geodataframe_iterfeatures_non_scalars`? this is already a rather long test method and it's nice to be testing specific bugs with their own asserts that can fail independently.